packages/fxa-auth-server/lib/account-delete.ts

```diff
@@ -2,39 +2,55 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { CloudTasksClient } from '@google-cloud/tasks';
+import {
+  deleteAllPayPalBAs,
+  getAllPayPalBAByUid,
+} from 'fxa-shared/db/models/auth';
 import { StatsD } from 'hot-shots';
 import { Container } from 'typedi';
+
+import { CloudTasksClient } from '@google-cloud/tasks';
+import * as Sentry from '@sentry/node';
+
+import { ConfigType } from '../config';
 import DB from './db/index';
+import { ERRNO } from './error';
 import OAuthDb from './oauth/db';
 import { AppleIAP } from './payments/iap/apple-app-store/apple-iap';
 import { PlayBilling } from './payments/iap/google-play/play-billing';
 import { PayPalHelper } from './payments/paypal/helper';
 import { StripeHelper } from './payments/stripe';
+import { StripeFirestoreMultiError } from './payments/stripe-firestore';
+import pushBuilder from './push';
 import pushboxApi from './pushbox';
 import { accountDeleteCloudTaskPath } from './routes/cloud-tasks';
-import { AccountDeleteReasons, AppConfig, AuthLogger } from './types';
-
-import {
-  deleteAllPayPalBAs,
-  getAllPayPalBAByUid,
-} from 'fxa-shared/db/models/auth';
-import { ConfigType } from '../config';
-import * as Sentry from '@sentry/node';
-import { StripeFirestoreMultiError } from './payments/stripe-firestore';
+import { AppConfig, AuthLogger, AuthRequest } from './types';
 
 type FxaDbDeleteAccount = Pick<
   Awaited<ReturnType<ReturnType<typeof DB>['connect']>>,
-  'deleteAccount' | 'accountRecord' | 'account'
+  'deleteAccount' | 'accountRecord' | 'account' | 'devices'
 >;
 type OAuthDbDeleteAccount = Pick<typeof OAuthDb, 'removeTokensAndCodes'>;
 type PushboxDeleteAccount = Pick<
   ReturnType<typeof pushboxApi>,
   'deleteAccount'
 >;
+type PushForDeleteAccount = Pick<
+  ReturnType<typeof pushBuilder>,
+  'notifyAccountDestroyed'
+>;
+type Log = AuthLogger & { activityEvent: (data: Record<string, any>) => void };
 
-export type ReasonForDeletion = (typeof AccountDeleteReasons)[number];
-export type AccountDeleteOptions = { notify?: () => Promise<void> };
+export const enum ReasonForDeletion {
+  UserRequested = 'fxa_user_requested_account_delete',
+  Unverified = 'fxa_unverified_account_delete',
+  Cleanup = 'fxa_cleanup_account_delete',
+}
+export const ReasonForDeletionValues = [
+  'fxa_user_requested_account_delete',
+  'fxa_unverified_account_delete',
+  'fxa_cleanup_account_delete',
```

Comment:
These values are already defined above.  It would be nice to keep them in one place so that the next time someone add a new one they won't need to remember to add it in more than one place.